packages/cli/src/active-executions.ts

```diff
@@ -163,6 +175,12 @@ export class ActiveExecutions {
 			promise.reject(reason);
 		}
 
+		// FIXME: Option 2: Don't clean up the execution. Because we cancelled it
```

Comment:
This function can probably be removed, because cancelling the promises is probably not correct, given that cancelling an execution will still resolve the promise returned from it:

https://github.com/n8n-io/n8n/blob/d61d563f48c06c889bef23021ab0295704ca1f50/packages/core/src/WorkflowExecute.ts#L857-L857

And in case we want to reject this, then the `WorkflowRunner` should do it after the execution promise has been resolved and it checked that it resolved early because it was cancelled.